tests/snippets/fstrings.py

```diff
@@ -16,6 +16,9 @@
 assert f'{16:0>+#10x}' == '00000+0x10'
 assert f"{{{(lambda x: f'hello, {x}')('world}')}" == '{hello, world}'
 
+spec = "0>+#10x"
+assert f"{16:{spec}}{foo}" == '00000+0x10bar'
+
```

Comment:
Could you add some TypeError tests here for formatting with `int`s as specs or similar?

Reply:
Could you be more specific?

Reply:
e.g. `assert_raises(TypeError, lambda: f"{123:{45}}")`

Reply:
> e.g. `assert_raises(TypeError, lambda: f"{123:{45}}")`

This is actually a valid format string.
I added a TODO in the tests for an invalid one. This currently does not fail but it is not related to this change.